core/src/Drawer.html

```diff
@@ -0,0 +1,226 @@
+    });
+
+    const prepareNodeData = async path => {
+        const pathUrlRaw =
+            path && path.length ? GenericHelpers.getPathWithoutHash(path) : '';
+        const params = RoutingHelpers.parseParams(pathUrlRaw.split('?')[1]);
+        nodeParams = RoutingHelpers.getNodeParams(params);
+        const dataFromPath = await Navigation.extractDataFromPath(path);
+        nodeObject = dataFromPath.nodeObject;
+        pathData = dataFromPath.pathData;
+
+        isDataPrepared = true;
+    };
+
+    const getNode = async path => {
+        if (iframeCreated) {
+            return;
+        }
+        if (isDataPrepared) {
+            const iframe = await createIframeDrawer(nodeObject.viewUrl, {
+                context: pathData.context,
+                pathParams: pathData.pathParams,
+                nodeParams
+            });
+            dispatch('iframeCreated', {
+                drawerIframe: iframe,
+                drawerIframeData: { ...pathData, nodeParams }
+            });
+            iframeCreated = true;
+        } else {
+            await prepareNodeData(path);
+        }
+    };
+    const createIframeDrawer = async (viewUrl, componentData) => {
+        await setDrawerSize();
+        if (viewUrl) {
+            viewUrl = RoutingHelpers.substituteViewUrl(viewUrl, componentData);
+        }
+
+        const iframe = await IframeHelpers.createIframe(
+            viewUrl,
+            undefined,
+            nodeObject,
+            'drawer'
+        );
+
+        const iframeCtn = document.querySelector('.iframeDrawerCtn');
+        iframeCtn.appendChild(iframe);
+        return iframe;
+    };
+
+    async function setDrawerSize() {
+        let styleSettings = '';
+        const elem = document.getElementsByClassName('drawer');
+        if (drawerSettings.width) {
+            styleSettings = `width:${drawerSettings.width};`;
+        }
+        if (drawerSettings.height) {
+            styleSettings += `height:${drawerSettings.height};`;
+        }
+        if (drawerSettings.position === 'left') {
+            styleSettings += `left:0;right:auto;top:0;`;
+        } else if (drawerSettings.position === 'right') {
+            styleSettings += `left:auto;right:0;top:0;`;
+        } else if (drawerSettings.position === 'bottom') {
+            styleSettings += `bottom:0;left:auto;right:auto;top:auto`;
+        } else if (drawerSettings.position === 'top') {
+            styleSettings += `top:0;left:auto;right:auto;`;
+        }
+        elem[0].setAttribute('style', styleSettings);
+    }
+
+    afterUpdate(() => {
+        getNode(nodepath);
+    });
+</script>
+<style>
+    .drawer {
+        top: 44px;
+        height: 100%;
+        width: 250px;
+        position: fixed;
+        overflow-y: auto;
+        left: auto;
+        right: 0;
+        /* flex: 1 0 auto; */
+        /* display: flex; */
+        /* outline: 0;
+        z-index: 1200; */
+        /* flex-direction: column; */
```

Comment:
clean up if not needed?

Reply:
It's a left over. I delete the whole file it is not used since the Modal component is used for the drawer.